src/hotspot/share/runtime/safefetch_static.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 SAP SE. All rights reserved.
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_RUNTIME_SAFEFETCH_STATIC_HPP
+#define SHARE_RUNTIME_SAFEFETCH_STATIC_HPP
+
+#include "utilities/globalDefinitions.hpp"
+
+extern "C" int SafeFetch32(int* adr, int errValue);
+
+#ifdef _LP64
+extern "C" intptr_t SafeFetchN(intptr_t* adr, intptr_t errValue);
+#else
+inline intptr_t SafeFetchN(intptr_t* adr, intptr_t errValue) {
+  return SafeFetch32(adr, errValue);
+}
+#endif // _LP64
+
+inline bool CanUseSafeFetch32() { return true; }
+inline bool CanUseSafeFetchN()  { return true; }
```

Comment:
AFAICS all CanUseSafeFetchXX always return true and there are few calls of these functions. It's worth to remove them completely then.